pkg/query-service/app/integrations/thirdPartyApi/translator.go

```diff
@@ -0,0 +1,337 @@
+package thirdPartyApi
+
+import (
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"net"
+)
+
+var defaultStepInterval int64 = 60
+
+func FilterResponse(results []*v3.Result) []*v3.Result {
+	filteredResults := make([]*v3.Result, 0, len(results))
+
+	for _, res := range results {
+		if res.Table == nil {
+			continue
+		}
+		filteredRows := make([]*v3.TableRow, 0, len(res.Table.Rows))
+		for _, row := range res.Table.Rows {
+			if row.Data != nil {
+				if domainVal, ok := row.Data["net.peer.name"]; ok {
+					if domainStr, ok := domainVal.(string); ok {
+						if net.ParseIP(domainStr) != nil {
+							continue
+						}
+					}
+				}
+			}
+			filteredRows = append(filteredRows, row)
+		}
+		res.Table.Rows = filteredRows
+
+		filteredResults = append(filteredResults, res)
+	}
+
+	return filteredResults
+}
+
+func getFilterSet(existingFilters []v3.FilterItem, apiFilters v3.FilterSet) []v3.FilterItem {
+	if len(apiFilters.Items) != 0 {
+		existingFilters = append(existingFilters, apiFilters.Items...)
+	}
+	return existingFilters
+}
+
+func getGroupBy(existingGroupBy []v3.AttributeKey, apiGroupBy []v3.AttributeKey) []v3.AttributeKey {
+	if len(apiGroupBy) != 0 {
+		existingGroupBy = append(existingGroupBy, apiGroupBy...)
+	}
+	return existingGroupBy
+}
+
+func BuildDomainList(thirdPartyApis *ThirdPartyApis) (*v3.QueryRangeParamsV3, error) {
+
+	unixMilliStart := thirdPartyApis.Start
+	unixMilliEnd := thirdPartyApis.End
+
+	builderQueries := make(map[string]*v3.BuilderQuery)
+
+	builderQueries["endpoints"] = &v3.BuilderQuery{
+		QueryName:         "endpoints",
+		DataSource:        v3.DataSourceTraces,
+		StepInterval:      defaultStepInterval,
+		AggregateOperator: v3.AggregateOperatorCountDistinct,
+		AggregateAttribute: v3.AttributeKey{
+			Key:      "http.url",
+			DataType: v3.AttributeKeyDataTypeString,
+			Type:     v3.AttributeKeyTypeTag,
+		},
+		TimeAggregation:  v3.TimeAggregationCountDistinct,
+		SpaceAggregation: v3.SpaceAggregationSum,
+		Filters: &v3.FilterSet{
+			Operator: "AND",
+			Items:    getFilterSet([]v3.FilterItem{}, thirdPartyApis.Filters),
+		},
+		Expression: "endpoints",
+		GroupBy: getGroupBy([]v3.AttributeKey{
+			{
+				Key:      "net.peer.name",
+				DataType: v3.AttributeKeyDataTypeString,
```

Comment:
you are creating columns for these right ? you will have to make that change here.

and also set isColumn= true.

Please complete the changes in https://github.com/SigNoz/signoz-otel-collector/pull/542 and make the updates to all of these queries.

Reply:
for now I want to keep the changes as it is, once the migration changes are merged and a collector is released, I'd add a followup PR to update the attributes.